Fix remote docker exec from Windows build of Emacs

p3r7 opened the pull request:
When runing the windows build of Emacs, but accessing a remote server hosting docker, commands would fail as `docker-utils-shell-command-to-string` would default to `cmdproxy.exe`.

This small fix addresses the issue.

Silex commented:
Thanks! Can you use `let` instead of `let*` here? Also the commit message being "Use cmdproxy.exe only when on local filesystems"

p3r7 commented:
Oh, yeah, the `let*` was a leftover from an initial version where I split the condition in two lines.

I also replaced the commit message with an amend. Would that be fine with you? I'm not so much a git wizard.

Silex commented:
I was able to squash all your commits in one before merging because github makes that possible, but for the next time I prefer directly just one commit. You can look how to rebase your commit and "fixup" them.

Anyway, thanks a lot! :+1:

p3r7 commented:
> You can look how to rebase your commits and "fixup" them.

Thanks for the tip!

> Anyway, thanks a lot!

You're welcome :beer:
